### ISSUE TITLE
`.case` allows properties to be return that aren't in the declared `initialState`

### ISSUE THREAD

**Issue by dcurletti** [aikoven/typescript-fsa#63]
```es6
interface Foo {
  bar: boolean
}

const INITIAL_STATE: Foo = {
  bar: true
}

const reducer = reducerWithInitialState(INITIAL_STATE).case(someAction, (state, payload) => {
  return {bar: true, SHOULDNT_ALLOW_THIS: true};
})
```
Following code typechecks, when `SHOULDNT_ALLOW_THIS` should throw a typecheck error.

**Comment by aikoven** [comment]
I'm not the author of `typescript-fsa-reducers`, you should probably file this issue in [its repo](https://github.com/dphilipson/typescript-fsa-reducers).

**Comment by dcurletti** [comment]
@aikoven whoops, sorry about that!
